packages/teams-js/src/public/visualMedia.ts

```diff
@@ -0,0 +1,307 @@
+     * @hidden
+     * The visualMedia source
+     */
+    source: Source.Gallery;
+  }
+  /**
+   * The required value of the visualMedia files from camera
+   * @beta
+   */
+  export interface CameraProps {
+    /**
+     * @hidden
+     * The visualMedia source
+     */
+    source: Source.Camera;
+    /**
+     * @hidden
+     * Optional; indicate if user is allowed to move between front and back camera
+     * Default value is FrontOrRear
+     */
+    cameraRestriction?: CameraRestriction;
+  }
+
+  /**
+   * Indicate if user is allowed to move between front and back camera or stay in front/back camera only
+   * @beta
+   */
+  export enum CameraRestriction {
+    /** User can move between front and back camera */
+    FrontOrRear = 1,
+    /** User can only use the front camera */
+    FrontOnly = 2,
+    /** User can only use the back camera */
+    RearOnly = 3,
+  }
+
+  /**
+   * Enum for file formats supported
+   * @beta
+   */
+  export enum FileFormat {
+    /** Base64 encoding */
+    Base64 = 'base64',
+    /** File id */
+    ID = 'id',
+  }
+  /**
+   * Specifies the image source
+   * @beta
+   */
+  export enum Source {
+    /** visual media source is camera. */
+    Camera = 1,
+    /** visual media source is gallery. */
+    Gallery = 2,
+  }
+  /**
+   * Specifies the image output formats.
+   * @beta
+   */
+  export enum ImageOutputFormats {
+    /** Outputs image.  */
+    Image = 1,
+    /** Outputs pdf. */
+    PDF = 2,
+  }
+
+  /**
+   * VisualMediaFile object that can be used to represent image or video
+   *
+   * @beta
+   */
+  export class VisualMediaFile {
+    /**
+     * Size of the file in KB
+     */
+    public size: number;
+
+    /**
+     * Name of the file
+     */
+    public name: string;
+    /**
+     * A preview of the file which is a lightweight representation.
+     * In case of images this will be a thumbnail/compressed image in base64 encoding.
+     */
+    public preview: string;
+
+    /**
+     * The fileProps object can be used to represent either an image or video with base64 content, or a file ID that maps to the URI.
+     *
+     * @beta
+     */
+    public fileProps: EncodedFormattedProps | IDFormattedProps;
+  }
+
+  /**
+   * EncodedFormattedProps object that can be used to represent image or video with the base64 content.
+   *
+   * @beta
+   */
+  export class EncodedFormattedProps {
+    /**
+     * This is the base64 content of file.
+     * When format is base64 and app needs to use this directly in HTML tags, it should convert this to a data url.
+     */
+    public content: string;
+    /**
+     * Format of the file
+     */
+    public format: FileFormat.Base64;
+
+    /**
+     * MIME type. This can be used for constructing a data url, if needed.
+     */
+    public mimeType: string;
+  }
+
+  /**
+   * IDFormattedProps object that can be used to represent image or video with the id mapping to the URI.
+   *
+   * @beta
+   */
+  export class IDFormattedProps {
```

Comment:
I don't really know how these ids can be used by developers to retrieve the full image. We should probably have more documentation here around how to build the URI and any other detail they may need.

Reply:
Based on what we discussed last Tuesday, we will make the id field as hidden.

Reply:
It looks like there are no functions that take advantage of this id field (I believe we talked last week about having the id hidden so that those files could be passed to a function that then made a request to the host for the "full" file, which would mean that the base64 content on the image was the "thumbnail". It looks like your current iteration just has the full image in the content of VisualMediaFile, so is there any reason we need to make the id available to developers?

Reply:
Yes, agree. I removed the id field.